src/Compilers/CSharp/Portable/Symbols/TypeSymbolExtensions.cs

```diff
@@ -1333,36 +1336,38 @@ internal static bool IsCustomTaskType(this NamedTypeSymbol type, out NamedTypeSy
         /// </summary>
         internal static TypeSymbol NormalizeTaskTypes(this TypeSymbol type, CSharpCompilation compilation)
         {
-            NormalizeTaskTypesCore(compilation, ref type);
+            NormalizeTaskTypesInType(compilation, ref type);
             return type;
         }
 
         /// <summary>
         /// Replace Task-like types with Task types. Returns true if there were changes.
         /// </summary>
-        private static bool NormalizeTaskTypesCore(CSharpCompilation compilation, ref TypeSymbol type)
+        private static bool NormalizeTaskTypesInType(CSharpCompilation compilation, ref TypeSymbol type)
         {
             switch (type.Kind)
```

Comment:
An array cannot be task-like but it can contain task-like types.  I don't see the benefit of digging into pointers / error types here because they can't contain task-like types.  At least to my knowledge :) 

Maybe it's possible for pointers ... 

---
In reply to: [74289506](https://github.com/dotnet/roslyn/pull/12671#discussion_r74289506) [](ancestors = 74289506,74288645,74285338)